listen.js

```diff
@@ -1,47 +1,100 @@
     return pull(
-      ssb.messagesByType({ type: 'group/add-member', private: true, live: true }),
+      pullMany([
+        ssb.db.query(
+          where(
+            and(
+              isDecrypted('box2'),
+              type('group/add-member')
+            )
+          ),
+          dbLive({ old: true }),
+          toPullStream()
+        ),
+        pull(
+          ssb.db.reindexed(),
+          pull.filter((msg) => {
+            return msg.value.content.type === 'group/add-member'
+          })
+        )
+      ]),
       // NOTE this will run through all messages on each startup, which will help guarentee
       // all messages have been emitted AND processed
       // (same not true if we used a dummy flume-view)
       pull.filter(m => m.sync !== true),
       pull.filter(isAddMember),
       pull.unique('key')
       // NOTE we DO NOT filter our own messages out
-      // this is important for rebuilding indexes and keystore state if we have to restore our feed
+      // this is important for rebuilding indexes and keyring state if we have to restore our feed
     )
   },
   excludeMember (ssb) {
     return pull(
-      ssb.messagesByType({ type: 'group/exclude-member', private: true, live: true }),
+      pullMany([
+        ssb.db.query(
+          where(
+            and(
+              isDecrypted('box2'),
+              type('group/exclude-member')
+            )
+          ),
+          dbLive({ old: true }),
+          toPullStream()
+        ),
+        pull(
+          ssb.db.reindexed(),
+          pull.filter((msg) => {
+            return msg.value.content.type === 'group/exclude-member'
+          })
+        )
+      ]),
       pull.filter(m => m.sync !== true),
       pull.filter(isExcludeMember),
       pull.unique('key')
     )
   },
   poBox (ssb, emit) {
+    const { isUpdate: isPOBox } = new CRUT(ssb, poBoxSpec).spec
+
     pull(
-      ssb.messagesByType({ type: 'group/po-box', private: true, live: true }),
+      pullMany([
+        ssb.db.query(
+          where(
+            and(
+              isDecrypted('box2'),
+              type('group/po-box')
+            )
+          ),
+          dbLive({ old: true }),
+          toPullStream()
+        ),
+        pull(
+          ssb.db.reindexed(),
+          pull.filter((msg) => {
+            return msg.value.content.type === 'group/po-box'
+          })
+        )
+      ]),
       // NOTE this will run through all messages on each startup, which will help guarentee
       // all messages have been emitted AND processed
       // (same not true if we used a dummy flume-view)
       pull.filter(m => m.sync !== true),
       pull.filter(isPOBox),
       pull.unique('key'),
       // NOTE we DO NOT filter our own messages out
-      // this is important for rebuilding indexes and keystore state if we have to restore our feed
+      // this is important for rebuilding indexes and keyring state if we have to restore our feed
       pull.drain(emit)
```

Comment:
this function now breaks the pattern of the others - some emit, some return pull-stream.
# :shrug:

Reply:
another PR I think

Reply:
yeah this is a change from an older pr